### DISCUSSION TITLE
how to create index  on scalars field with go sdk?

### DISCUSSION THREAD

**Discussion by younger027** [milvus-io/milvus#28462] [Q&A and General discussion] [answered]
i use go sdk v2.3.1 to build index for vector and scalars filed. code like
`idx, err := entity.NewIndexIvfFlat(
		entity.L2, // metricType
		1024,      // nlist
	)
	if err != nil {
		logx.Error("fail to create ivf flat index parameter:", err.Error())
		return err
	}

	//vector index
	err = mClient.CreateIndex(ctx, collectionName, vectorFieldName, idx, false)
	if err != nil {
		logx.Error("fail to create index parameter:", err.Error())
		return err
	}

	//how to build index on scalars field
	mClient.CreateIndex(ctx, collectionName, "scalars", "???", true)`
question:
1.which ??? idx params should be use to build index on scalars field? i can not find func like "createIndexForScalars".
2.i search the other discussions know that we do not need build for scalars field. milvus also can query data fast. is that true?

**Comment by yhmo** [comment, marked as answer]
1. I think entity.NewScalarIndex() can work:  https://github.com/milvus-io/milvus-sdk-go/blob/d1df1d6b12f262652ebe0b454de0c612f1a3c1ad/entity/index_scalar.go#L13
2. in most cases, you don't need to create index for scalar field because the bottleneck is mainly in vectors distance computation.
